src/Stream/Partition.cpp

```diff
@@ -82,26 +84,22 @@ bool Partition::shouldStopBasedOnPollStatus(Kafka::PollStatus CStatus) {
 void Partition::pollForMessage() {
   auto Msg = ConsumerPtr->poll();
   switch (Msg.first) {
-  case Kafka::PollStatus::Message:
```

Comment:
I consider this change a bug. There are other metrics counters for keeping track of number of processed messages. I have created the relevant (code) "suggestions" for moving the messages received counter back to where it should be.

Reply:
I don't think it is too important if it is changed back, but I'd like to understand your comment if possible.
"There are other metrics counters for keeping track of number of processed messages", it isn't being used for that. The changed code looks functionally equivalent to me, it just avoids doing the comparison `Msg.first == Kafka::PollStatus::Message` twice. The only risk I see is that someone could now put some code between the poll and the call to `processMessage` which calls `return` and thus introduce a bug of the message not being counted.

Reply:
Moving `MessagesReceived++;` to after the `if (Msg.first == Kafka::PollStatus::Message) {` but outside of `processMessage()` would probably be better.

Reply:
Actually, saying "bug" in this case was not entirely correct. Still, my reasoning is that `MessagesReceived` indicated just that. In principle `shouldStopBasedOnPollStatus()` could return `false` (though it does not in the current implementation) which would then result in this metric being wrong. Furthermore, why move `MessagesReceived++;` it at all when the switch statement is used for the other counters as well? As for the extra comparison, without evidence I do not believe this to impact the performance in any measurable way.